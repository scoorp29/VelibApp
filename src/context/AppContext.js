import React, {useState, useEffect} from "react";
import {Animated} from "react-native";
import {get} from "../api/getAPIwithCache";
import {findCoordinates} from "../services/findCoordinatesServices";
import {cycleAnimation} from "../services/cycleAnimationServices";

export const AppContext = React.createContext(null);

export const AppProvider = props => {
    const [location, setLocation] = useState([]);
    const [data, setData] = useState([]);
    const [loading, setLoading] = useState(null);
    const [like,setLike] = useState([]);
    const [height] = useState(new Animated.Value(0.5));
    const [value, setValue] = React.useState('10000');

    const api = () => {
        let API_URL = "https://opendata.paris.fr/api/records/1.0/search/?dataset=velib-disponibilite-en-temps-reel&rows=1&facet=station_state&geofilter.distance=48.8534%2C2.3488%2C800";
        if (location.coords) {
            return API_URL = "https://opendata.paris.fr/api/records/1.0/search/?dataset=velib-disponibilite-en-temps-reel&rows=20&facet=station_state&geofilter.distance=" + location.coords.latitude + "%2C" + location.coords.longitude + "%2C" + value;
        }
    };

    console.log(data);

    const fetchData = networkPriority => {
        setLoading(true);
        get(api(), networkPriority).then(data => {
            setData(data);
            setLoading(false);
        });
    };

    useEffect(() => {
        cycleAnimation(height);
        findCoordinates(location, setLocation);
    }, []);

    useEffect(() => {
        if (location) {
            if(value) {
            fetchData(true);
            }
        }
        fetchData();
    }, [location, value]);

    return <AppContext.Provider value={{location, setLocation, data, setData, like, setLike, height, value, setValue}} {...props} />;
};
