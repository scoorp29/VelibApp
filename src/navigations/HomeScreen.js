import React from 'react';
import {Button, View, StyleSheet, Animated} from 'react-native';
import {AppContext} from "../context/AppContext";

const HomeScreen = (props) => {

    return (
        <View style={styles.container}>
                <AppContext.Consumer>
                    {context => (
                            <Animated.View style={[{flex: context.height, backgroundColor: '#f9b01f'}]}>
                                <View style={[styles.container2]}>
                                <Button
                                    color="#adb924"
                                    title="Velib Station"
                                    onPress={() => props.navigation.navigate('Stations')}
                                />
                                </View>
                            </Animated.View>
                    )}
                </AppContext.Consumer>
            <View style={[styles.container2, {backgroundColor: '#df2f8a'}]}>
                <Button
                    color="#15aac3"
                    title="Velib Maps"
                    onPress={() => props.navigation.navigate('Maps')}
                />
            </View>
        </View>
    )
};

export default HomeScreen;

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    container2: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#f9b01f',
    },
});