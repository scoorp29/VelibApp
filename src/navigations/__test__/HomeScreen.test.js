import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from "../HomeScreen";
import {AppProvider} from "../../context/AppContext";

test('renders correctly HommeScreen', () => {

    const tree = renderer.create(
        <AppProvider>
            <HomeScreen/>
        </AppProvider>
    ).toJSON();


    expect(tree).toMatchSnapshot();
});