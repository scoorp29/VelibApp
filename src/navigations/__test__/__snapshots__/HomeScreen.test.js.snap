// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly HommeScreen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#f9b01f",
        "flex": 0.5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#f9b01f",
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#adb924",
                },
              ]
            }
          >
            Velib Station
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f9b01f",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#df2f8a",
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#15aac3",
              },
            ]
          }
        >
          Velib Maps
        </Text>
      </View>
    </View>
  </View>
</View>
`;
